src/main.rs

```diff
@@ -112,20 +112,24 @@ fn exit_on_error<T>(log_errors: bool, inner: impl Fn() -> anyhow::Result<T>) ->
 
 /// helper function to determine whether to use colors based on the (passed) `--color` flag
 /// and the value of the `CARGO_TERM_COLOR` variable.
-fn configure_color(cli_choice: colorchoice_clap::Color) {
-    cli_choice.write_global();
-    // if it is still auto (i.e., user passed `--color=auto` or omitted the flag entirely),
-    // we check the env var.  otherwise, the flag takes precedence over whatever the env var is
-    if ColorChoice::global() == ColorChoice::Auto {
+fn configure_color(cli_choice: ColorChoice) {
+    // if the --color flag is explicitly set to something other than `auto`,
+    // this takes precedence over the environment variable, so we write that
+    if cli_choice != ColorChoice::Auto {
+        cli_choice.write_global();
+    } else {
```

Comment:
Shouldn't an _explicitly passed_ `--color=auto` CLI flag override whatever `CARGO_TERM_COLOR` is set to? I find this note a bit surprising.

Reply:
That's a good point, and in playing around with `cargo`, that's the behavior it does.  The `colorchoice_clap` crate doesn't currently distinguish between an explicitly-passed `--color=auto` and no color flag entirely.  I will try right now if it works with an `Option<color_choice::Color>` but if that doesn't play nice with `#[clap(flatten)]`, we might have to move back to the bespoke color configuration flag that I previously had.

Relevant tangent: In doing this, I discovered that the environment variable actually takes precedence over the CLI flag, unlike what we had previously done.  It's only been a little while since I added the `--color` flag in general; do we want to match cargo's behavior and read the environment variable over the flag?

Reply:
Oh that's interesting! Yes, let's match cargo's behavior then -- and ideally, let's document this if at all possible, since I imagine other people might find this a bit unexpected as well.

If we don't document it, I'd be shocked if we don't get a "bug report" on that behavior in the future.

Reply:
Oops, little bit of a false alarm - it looks like `cargo help` checks the `CARGO_TERM_COLOR` env var but not the `--color` flag - may be a tiny bug in cargo itself.  When I test with `CARGO_TERM_COLOR=env cargo check --color=cli`, `cli` overrides `env` if it is set. However, you're right that we still need to distinguish an explicit `--color=auto` from one that's not set - I'll work on that.